### PR TITLE
Logging useful information based on gate evaluation response

### PR DESCRIPTION
**Pull request by andrea-mosk**
### What and why?

Based on gate evaluation response returned, we log important information to customers. 

Example of logs:

Screenshot 2023-05-17 at 12 55 49

### Review checklist

- [x] Feature or bugfix MUST have appropriate tests (unit, integration)

### CODE REVIEW

File: src/commands/gate/renderer.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  INFO: 'ℹ️',
+}
+
+export const renderEvaluationResponse = (evaluationResponse: EvaluationResponse) => {
+  if (evaluationResponse.status.toLowerCase() === 'empty') {
+    return renderEmptyEvaluation()
+  }
+
+  let fullStr = ''
+  fullStr += chalk.green(`Successfully evaluated rules for the current pipeline.\n`)
```
Comment (juan-fernandez) [inline src/commands/gate/renderer.ts:20]: nitpick: if there are no variables in the template, there's no need to use "`" 
```suggestion
  fullStr += chalk.green('Successfully evaluated rules for the current pipeline.\n')
```

File: src/commands/gate/renderer.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
+import chalk from 'chalk'
+
+import {SpanTags} from '../../helpers/interfaces'
+
+import {EvaluationResponse, RuleEvaluation} from './interfaces'
+
+const ICONS = {
+  FAILED: '❌',
```
Comment (juan-fernandez) [inline src/commands/gate/renderer.ts:8]: nitpick: I see other usages of icons in this repo as
```suggestion
  FAILED: chalk.bold.red('❌'),
```
do we need that as well?

File: src/commands/gate/renderer.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+  fullStr += '\n'
+  fullStr += chalk.yellow('####### Rules evaluated #######\n')
+  evaluationResponse.rule_evaluations.forEach((ruleEvaluation) => (fullStr += renderRuleEvaluation(ruleEvaluation)))
+
+  return fullStr
+}
+
+export const renderEmptyEvaluation = (): string => {
+  return chalk.yellow(`No matching rules were found in Datadog for the current pipeline.\n`)
```
Comment (juan-fernandez) [inline src/commands/gate/renderer.ts:32]: nitpick
```suggestion
  return chalk.yellow('No matching rules were found in Datadog for the current pipeline.\n')
```

File: src/commands/gate/renderer.ts [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+  }
+
+  fullStr += `${chalk.yellow('Blocking')}: ${ruleEvaluation.is_blocking}\n`
+  fullStr += '\n'
+
+  return fullStr
+}
+
+export const renderDryRunEvaluation = (): string => {
+  return chalk.yellow(`Dry run mode is enabled. Not evaluating the rules.`)
```
Comment (juan-fernandez) [inline src/commands/gate/renderer.ts:65]: nitpick
```suggestion
  return chalk.yellow('Dry run mode is enabled. Not evaluating the rules.')
```

File: src/commands/gate/interfaces.ts [hunk 5]
```diff
@@ -1,13 +1,33 @@
...
 }
 
+export interface EvaluationResponsePayload {
+  data: {
+    attributes: EvaluationResponse
+  }
+}
+
+export interface EvaluationResponse {
+  status: string
```
Comment (juan-fernandez) [inline src/commands/gate/interfaces.ts:18]: could we have a union here? 
```javascript
type GateStatus = 'passed' | 'failed'
// ...
status: GateStatus
```

File: src/commands/gate/interfaces.ts [hunk 6]
```diff
@@ -1,13 +1,33 @@
...
+  rule_evaluations: RuleEvaluation[]
+}
+
+export interface RuleEvaluation {
+  rule_id: string
+  rule_name: string
+  status: string
```
Comment (juan-fernandez) [inline src/commands/gate/interfaces.ts:25]: same as above
```javascript
  type EvaluationStatus = 'passed' | 'failed' | 'warning' // just made up the statuses
  // ...
  status: EvaluationStatus
```

File: src/commands/gate/evaluate.ts [hunk 7]
```diff
@@ -58,19 +64,22 @@ export class GateEvaluateCommand extends Command {
...
 
-    // To be extended
-    try {
-      await api.evaluateGateRules(evaluateRequest, this.context.stdout.write.bind(this.context.stdout))
-    } catch (error) {
-      this.context.stderr.write(error)
-    }
+    // To be extended with retries, error handling, etc.
+    await api
+      .evaluateGateRules(evaluateRequest, this.context.stdout.write.bind(this.context.stdout))
```
Comment (juan-fernandez) [inline src/commands/gate/evaluate.ts:76]: this will eventually change the exit code based on the response, right? But that's a follow up PR I guess

**Review by juan-fernandez — APPROVED** [review]
LGTM, small nitpicks

File: src/commands/gate/renderer.ts (hunk 2 above)
Comment (andrea-mosk) [reply src/commands/gate/renderer.ts:8]: I don't think we need it, I copied this from the sarif/junitxml command

File: src/commands/gate/evaluate.ts (hunk 7 above)
Comment (andrea-mosk) [reply src/commands/gate/evaluate.ts:76]: Yup, that's the case. I will change this behavior in the next PRs

File: src/commands/gate/interfaces.ts (hunk 5 above)
Comment (andrea-mosk) [reply src/commands/gate/interfaces.ts:18]: Yeah looks good! Changed it

File: src/commands/gate/interfaces.ts (hunk 6 above)
Comment (andrea-mosk) [reply src/commands/gate/interfaces.ts:25]: Changed it as well

**Review by liashenko — APPROVED** [review]

**Pull request merged**